src/ui/views/PlayerStats.tsx

```diff
@@ -64,112 +65,44 @@ const PlayerStats = ({
 		},
 	});
 
-	const cols = getCols([
-		"Name",
-		"Pos",
-		"Age",
-		"Team",
-		...(season === "all" ? ["Season"] : []),
-		...stats.map(
-			stat => `stat:${stat.endsWith("Max") ? stat.replace("Max", "") : stat}`,
-		),
-	]);
+	const cols = config.columns;
 
-	if (statType === "shotLocations") {
-		cols[cols.length - 7].title = "M";
-		cols[cols.length - 6].title = "A";
-		cols[cols.length - 5].title = "%";
-	}
-
-	let sortCol = cols.length - 1;
+	let sortCol: string = cols[0].key ?? "col1";
+	let sortDir: SortOrder = "asc";
 	if (isSport("football")) {
 		if (statType === "passing") {
-			sortCol = 9;
+			sortCol = "stat:passYds";
+			sortDir = "desc";
 		} else if (statType === "rushing") {
-			sortCol = cols.length - 3;
+			sortCol = "stat:rusRecTD";
+			sortDir = "desc";
 		} else if (statType === "defense") {
-			sortCol = 16;
+			sortCol = "stat:defSk";
+			sortDir = "desc";
 		} else if (statType === "kicking") {
-			sortCol = cols.length - 11;
+			sortCol = "stat:fgPct";
+			sortDir = "desc";
 		} else if (statType === "returns") {
-			sortCol = 12;
+			sortCol = "stat:krYds";
+			sortDir = "desc";
 		}
 	}
 
 	const rows = players.map(p => {
-		let pos;
-		if (Array.isArray(p.ratings) && p.ratings.length > 0) {
-			pos = p.ratings.at(-1).pos;
-		} else if (p.ratings.pos) {
-			pos = p.ratings.pos;
-		} else {
-			pos = "?";
-		}
-
-		// HACKS to show right stats, info
-		let actualAbbrev;
-		let actualTid;
 		if (season === "career") {
 			p.stats = p.careerStats;
-			actualAbbrev = p.abbrev;
-			actualTid = p.tid;
 			if (playoffs === "playoffs") {
 				p.stats = p.careerStatsPlayoffs;
 			}
-		} else {
-			actualAbbrev = p.stats.abbrev;
-			actualTid = p.stats.tid;
 		}
-
-		const statsRow = stats.map(stat =>
-			formatStatGameHigh(p.stats, stat, statType),
-		);
-
-		const key = season === "all" ? `${p.pid}-${p.stats.season}` : p.pid;
-
 		return {
-			key,
-			data: [
```

Comment:
We lost all the special stuff here, right? Use of nameAbbrev, conditional setting of season, custom sort/search values, ageAtDeath rather than just age, conditional season column...

Reply:
Related to this... there are several tables that only display a column sometimes, based on some input condition. How should that be handled when columns are customized?

Reply:
![image](https://user-images.githubusercontent.com/45333542/151425073-4504ae74-abfd-45aa-acaa-77957c3d8e3f.png)
These drop downs should still work? what is the nameAbbrev and custom search ?

Reply:
nameAbbbrev displays only the initials of the first name, because this is a wide table. Arguably there's no need for it. But other stuff in various places there is need for, like setting the season passed to PlayerNameLabels which is used to set the season of the player stats/ratings popup.

Plus the other stuff I mentioned. I'm talking about the functionality of the table, not the dropdowns at the top. Like look what happens when a dead player appears when it's showing career totals. Or when you switch to "all seasons".